Templates/BaseGame/game/core/gui/scripts/canvas.tscript

```diff
@@ -202,27 +209,34 @@ function GuiCanvas::checkCanvasRes(%this, %mode, %deviceId, %deviceMode, %startu
    return false;
 }
 
-// Find the best video mode setting for the device and display mode
+// Find the best video mode setting for the device and display mode.
+// "Best" is the largest resolution that will fit at highest refresh rate.
 function GuiCanvas::getBestCanvasRes(%this, %deviceId, %deviceMode)
 {
    if (%deviceMode == $Video::ModeWindowed)
       %deviceRect = getWords(%this.getMonitorUsableRect(%deviceId), 2);
    else
       %deviceRect = getWords(%this.getMonitorRect(%deviceId), 2);
 
+   %bestRes = "";
    %resCount = %this.getModeCount();
    for (%i = %resCount - 1; %i >= 0; %i--)
    {
       %testRes = %this.getMode(%i);
       %resX = getWord(%testRes, $WORD::RES_X);
       %resY = getWord(%testRes, $WORD::RES_Y);
+      %rate = getWord(%testRes, $WORD::REFRESH);
 
-      if ((%resX > %deviceRect.x) || (%resY > %deviceRect.y))
+      if ((%resX > %deviceRect.x) || (%resY > %deviceRect.y) ||
+         (%resX < $Video::minimumXResolution) || (%resY < $Video::minimumYResolution))
          continue;
 
-      return %testRes;
+      if (((%bestRes $= "") || (%resX > getWord(%bestRes, $WORD::RES_X)) ||
+            (%resY > getWord(%bestRes, $WORD::RES_Y))) ||
+         ((%resX == getWord(%bestRes, $WORD::RES_X)) && (%resY == getWord(%bestRes, $WORD::RES_Y)) &&
+            (%rate > getWord(%bestRes, $WORD::REFRESH))))
+         %bestRes = %testRes;
```

Comment:
This is... This is fairly difficult to follow..

So:
If bestRes is not set OR resX is greater than the bestRes OR resY is greater than bestRes OR (resX is the same as bestRes AND resY is the same as bestRes AND rate is higher) then testRes = bestRes 

Could this be:

```suggestion
      %totalRes = %resX * %resY;
      %totalBestRest = getWord(%bestRes, $WORD::RES_X) * getWord(%bestRes, $WORD::RES_Y);
      if (%totalRes < %totalBestRes) continue;
      if (%totalRes == %totalBestRes && %rate <= getWord(%bestRes, $WORD::REFRESH)) continue;
      %bestRes = %testRes;
```

Reply:
Hesitant to add resolution checks in the form of %totalRes = %resX * %resY; in case that plays up for folks trying to add phone et al support down the line. we might not have plans to pursue it at time of writing, but lets not make headaches for future-us, eh?